core/src/test/java/org/apache/iceberg/rest/requests/TestUpdateNamespacePropertiesRequest.java

```diff
@@ -0,0 +1,162 @@
+        "A JSON request with an invalid type for one of the fields should fail to parse",
+        JsonProcessingException.class,
+        () -> deserialize(jsonInvalidTypeOnRemovalField).validate()
+    );
+
+    String jsonInvalidTypeOnUpdatesField =
+        "{\"removals\":[\"foo\":\"bar\"],\"updates\":[\"owner\"]}";
+    AssertHelpers.assertThrows(
+        "A JSON value with an invalid type for one of the fields should fail to parse",
+        JsonProcessingException.class,
+        () -> deserialize(jsonInvalidTypeOnUpdatesField).validate()
+    );
+
+    // Valid top-level (array) types, but at least one entry in the list is not the expected type
+    // NOTE: non-string values that are integral types will still parse into a string list.
+    //    e.g. { removals: [ "foo", "bar", 1234 ] } will parse correctly.
+    String invalidJsonWrongTypeInRemovalsList =
+        "{\"removals\":[\"foo\",\"bar\", {\"owner\": \"Hank\"}],\"updates\":{\"owner\":\"Hank\"}}";
+    AssertHelpers.assertThrows(
+        "A JSON value with an invalid type inside one of the list fields should fail to parse",
+        JsonProcessingException.class,
+        () -> deserialize(invalidJsonWrongTypeInRemovalsList).validate()
+    );
+
+    // All fields are valid types, but all are empty
+    String jsonWithAllFieldsAsEmptyCollection =
+        "{\"removals\":[],\"updates\":{}}";
+    AssertHelpers.assertThrows(
+        "A JSON value with empty collections for all fields should fail to parse and validate",
+        IllegalArgumentException.class,
+        "Cannot create a request to update a namespace's properties without specifying any props to update or remove",
+        () -> deserialize(jsonWithAllFieldsAsEmptyCollection).validate()
+    );
+
+    String emptyJson = "{}";
+    AssertHelpers.assertThrows(
+        "An empty JSON should fail to parse",
+        IllegalArgumentException.class,
+        "Cannot create a request to update a namespace's properties without specifying any props to update or remove",
+        () -> deserialize(emptyJson).validate()
+    );
+
+    String nullJson = null;
+    AssertHelpers.assertThrows(
+        "A null JSON should fail to parse",
+        IllegalArgumentException.class,
+        () -> deserialize(nullJson).validate()
+    );
+  }
+
+  @Override
+  public String[] allFieldsFromSpec() {
+    return new String[] { "updates", "removals" };
+  }
+
+  @Override
+  public UpdateNamespacePropertiesRequest createExampleInstance() {
+    return UpdateNamespacePropertiesRequest.builder()
+        .updateAll(UPDATES)
+        .removeAll(REMOVALS)
+        .build();
+  }
+
+  @Override
+  public void assertEquals(UpdateNamespacePropertiesRequest actual, UpdateNamespacePropertiesRequest expected) {
+    Assert.assertEquals("Properties to update should be equal", actual.updates(), expected.updates());
+    Assert.assertEquals("Properties to remove should be equal", actual.removals(), expected.removals());
```

Comment:
I think this should build a set from both removals lists. Otherwise, this could be flaky because the builder users a set to accumulate removals and doesn't provide order guarantees.

Reply:
Changed to use a Set. I also changed some other list comparison ones to use a set as well.